glib/Marshaller.cs

```diff
@@ -88,19 +88,19 @@ static string Utf8PtrToStringFast (IntPtr ptr, int len)
 			}
 		}
 
-		static bool hasFastGetStringOverload = typeof (System.Text.Encoding).GetMethod ("GetString", new [] { typeof (byte*), typeof (int) }) != null;
 		public static string Utf8PtrToString (IntPtr ptr) 
 		{
 			if (ptr == IntPtr.Zero)
 				return null;
 
 			int len = (int) (uint)glibsharp_strlen (ptr);
-			if (hasFastGetStringOverload)
-				return Utf8PtrToStringFast (ptr, len);
-
+#if HAVE_NET_4_6
```

Comment:
The diff is still here since you commented on the define, see below, the Utf8PtrStringFast is inlined now under the same HAVE_NET_4_6.

This way, you can use [unifdef](http://dotat.at/prog/unifdef/) to remove the segments.